src/Storages/StoragePostgreSQL.cpp

```diff
@@ -456,15 +458,24 @@ SinkToStoragePtr StoragePostgreSQL::write(
     return std::make_shared<PostgreSQLSink>(metadata_snapshot, pool->get(), remote_table_name, remote_table_schema, on_conflict);
 }
 
-StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, bool require_table)
+StoragePostgreSQL::Configuration StoragePostgreSQL::processNamedCollectionResult(const NamedCollection & named_collection, PostgreSQLSettings& storage_settings, bool require_table)
 {
     StoragePostgreSQL::Configuration configuration;
+
+    ValidateKeysMultiset<ExternalDatabaseEqualKeysSet> optional_arguments = {"replace_query", "on_duplicate_clause", "addresses_expr", "host", "hostname", "port"};
+    auto postgresql_settings = storage_settings.all();
+    for (const auto & setting : postgresql_settings)
+    {
+        const auto & setting_name = setting.getName();
+	if (named_collection.has(setting_name))
+            storage_settings.set(setting_name, named_collection.get<String>(setting_name));
+    }
     ValidateKeysMultiset<ExternalDatabaseEqualKeysSet> required_arguments = {"user", "username", "password", "database", "db"};
     if (require_table)
         required_arguments.insert("table");
 
     validateNamedCollection<ValidateKeysMultiset<ExternalDatabaseEqualKeysSet>>(
-        named_collection, required_arguments, {"schema", "on_conflict", "addresses_expr", "host", "hostname", "port", "use_table_cache"});
+        named_collection, required_arguments, {"schema", "on_conflict", "addresses_expr", "host", "port", "use_table_cache"});
```

Comment:
But after you returned optional arguments back here (in https://github.com/ClickHouse/ClickHouse/pull/52711/commits/aeb594e6f067fb0d20224e136109e60db72a2b0d) there is no point in the list you added:  `optional_arguments`:
https://github.com/ClickHouse/ClickHouse/blob/6b330e30b025291e4a114d6ba19472eea658680c/src/Storages/StorageMySQL.cpp#L262
because it is not used any more in keys validation, so the code in lines 566-470  is also pointless as the arguments will not pass keys validation.
As I wrote in https://github.com/ClickHouse/ClickHouse/pull/52711#discussion_r1282957408, you need to put them in line 465 instead (the list of optional argument in line 465 is currently incorrect as it was taken from MySQL)